go-controller/pkg/clustermanager/node/node_allocator.go

```diff
@@ -380,7 +394,7 @@ func (na *NodeAllocator) HandleDeleteNode(node *corev1.Node) error {
 		return nil
 	}
 
-	if na.hasNodeSubnetAllocation() {
+	if na.hasNodeSubnetAllocation() || na.hasHybridOverlayAllocationUnmanaged() {
 		na.clusterSubnetAllocator.ReleaseAllNetworks(node.Name)
```

Comment:
does this mechanism of releasing subnets work for subnets that may have different sizes than the cluster subnets?

Reply:
Yes, `ReleaseAllNetworks` can release all overlapped nodeSubnets occupied by one hybrid overlay node.